include/OpenSpaceToolkit/Astrodynamics/Conjunction/Generator/Generator.hpp

```diff
@@ -0,0 +1,169 @@
+#include <OpenSpaceToolkit/Astrodynamics/Solver/TemporalConditionSolver.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State.hpp>
+
+namespace ostk
+{
+namespace astrodynamics
+{
+namespace conjunction
+{
+
+using ostk::core::container::Array;
+using ostk::core::type::Real;
+
+using ostk::physics::time::Duration;
+using ostk::physics::time::Instant;
+using ostk::physics::time::Interval;
+
+using ostk::astrodynamics::conjunction::CloseApproach;
+using ostk::astrodynamics::solver::TemporalConditionSolver;
+using ostk::astrodynamics::Trajectory;
+using ostk::astrodynamics::trajectory::State;
+
+#define DEFAULT_STEP_SIZE Duration::Seconds(15.0)
+#define DEFAULT_TOLERANCE Duration::Milliseconds(1.0)
+
+/// @brief Find close approaches to a reference trajectory
+///
+/// @details This class finds close approach events between a reference trajectory and other object trajectories
+/// over a specified time interval. It uses a temporal condition solver to identify time periods when objects
+/// are approaching and then determines the exact time of closest approach.
+class Generator
+{
+   public:
+    /// @brief Constructor
+    ///
+    /// @code{.cpp}
+    ///              Trajectory referenceTrajectory = { ... } ;
+    ///              Duration stepSize = Duration::Seconds(15.0) ;
+    ///              Duration tolerance = Duration::Milliseconds(1.0) ;
+    ///              Generator generator = { referenceTrajectory, stepSize, tolerance };
+    /// @endcode
+    ///
+    /// @param aReferenceTrajectory The reference trajectory for which to find close approaches (Object 1)
+    /// @param aSearchStepSize The step size to use during the close approach search
+    /// @param aSearchTolerance The tolerance to use during the close approach search
+    Generator(
+        const Trajectory& aReferenceTrajectory,
+        const Duration& aSearchStepSize = DEFAULT_STEP_SIZE,
+        const Duration& aSearchTolerance = DEFAULT_TOLERANCE
+    );
+
+    /// @brief Check if the generator is defined
+    ///
+    /// @code{.cpp}
+    ///              Generator generator = { ... } ;
+    ///              generator.isDefined() ;
+    /// @endcode
+    ///
+    /// @return True if generator is defined
+    bool isDefined() const;
+
+    /// @brief Get the reference trajectory
+    ///
+    /// @code{.cpp}
+    ///              Generator generator = { ... } ;
+    ///              Trajectory referenceTrajectory = generator.getReferenceTrajectory() ;
+    /// @endcode
+    ///
+    /// @return The reference trajectory
+    Trajectory getReferenceTrajectory() const;
+
+    /// @brief Get the search step size
+    ///
+    /// @code{.cpp}
+    ///              Generator generator = { ... } ;
+    ///              Duration stepSize = generator.getSearchStepSize() ;
+    /// @endcode
+    ///
+    /// @return The search step size
+    Duration getSearchStepSize() const;
```

Comment:
rename to step to align with generator.
```suggestion
    Duration getStep() const;
```